src/System.Private.CoreLib/src/System/Threading/LowLevelSpinWaiter.cs

```diff
@@ -51,8 +51,15 @@ public bool SpinWaitForCondition(Func<bool> condition)
                     // prior to that threshold would not help other threads make progress
                     for (int spinIndex = processorCount > 1 ? 0 : SpinYieldThreshold; spinIndex < SpinCount; ++spinIndex)
                     {
-                        // The caller should check the condition in a fast path before calling this method, so wait first
-                        Wait(spinIndex);
+                        if (processorCount > 1)
+                        {
+                            // The caller should check the condition in a fast path before calling this method, so wait first
+                            Wait(spinIndex, SpinSleep0Threshold);
+                        }
+                        else
+                        {
+                            RuntimeThread.Yield();
```

Comment:
The `processorCount <= 1` case is already covered in the initial spin index. It would be a bit cleaner to change the use of `SpinYieldThreshold` above to `SpinSleep0Threshold` (then `SpinYieldThreshold` would not be used anymore and can be removed), and revert this block to what it was before.

Reply:
It is not covered because `Wait` method now also spins on every second iteration. A better solution would be to keep spin counter inside `LowLevelSpinWaiter`, but that's refactoring I would like to do separately.

Reply:
Perhaps `Wait` could check `s_processorCount` and sleep instead of spin, instead of checking the proc count at the caller side

Reply:
Yes, that's what I would want to do... but the initialization of the static variables and reading of the processor count is kinda funky there. I wasn't sure if it's safe to change it or if the code is possibly reached from some early static constructor initialization and it would result in some dangerous dependency.

Reply:
Oh ok, yea the wait subsystem and stuff it uses are used in lazy class construction. It would work to change the `Initialize` method into a static method (maybe rename to `EnsureInitialized` or something like that) and call it before any spin-wait loops that use this type. Not sure if it's the best solution. I think I had considered attributing this class with `EagerStaticClassConstruction` but maybe there was some issue with it, I don't remember the details.

Reply:
I tried to reorganize it a little in the last commit (https://github.com/dotnet/corert/pull/6955/commits/cffc32dcd9da2ed7449cb571b590cb0b775435d4). It's not perfect, but it's a bit more readable. I changed it to use `PlatformHelper.ProcessorCount` which is supposed to be respond to change in number of processors, but still do reasonable caching that works with the static constructor initialization.

Reply:
Damn, looks like it broke the universe. I'll have to debug it on my other machine tomorrow... Reverting for now.